semgrep/semgrep/core_exception.py

```diff
@@ -56,27 +59,38 @@ def from_json(  # type: ignore
         start_pos = Position(start["line"], start["col"])
         end_pos = Position(end["line"], end["col"])
 
+        # Semgrep-Core caches TimeoutErrors as FatalErrors
+        # Hack to treat these as TimeoutErrors
+        check_id = json_obj["check_id"]
+        if check_id == "FatalError" and "Timeout" in json_obj["extra"]["message"]:
+            check_id = "Timeout"
+
         return cls(
-            json_obj["check_id"],
+            check_id,
```

Comment:
how do you feel about `json_obj.get("check_id", rule_id)`?

Reply:
This is needed for the above hack 😢